gives deck crew helmets ear protection, allows helmets to have volume_multiplier

gy1ta23 opened the pull request:
🆑 Gy1ta23
rscadd: helmets can do volume multipliers, deck crew helmets now muffle sound
/🆑

All of this is done with the blessing of Joey, who didn't know that earmuffs actually muffled sound, nor that you could muffle sound via ear clothing at all. Love you, Joey 😘
Taza helped me do all the code for letting helmets use the volume multipliers. Very essential, thank you

Now, anyways:

**A SPECTRE HAUNTS THIS CODEBASE.**

THE SPECTRE OF SERVICE-INDUCED HEARING LOSS. Do you think the SCG VA cares that you have tinnitus from working the deck? Of course not! You're issued ear-pro as part of your job, aren't you? But that's just the thing! Joey advertised his crew helmet as sound-protecting, but just like 3M Combat Arms Earplugs from 2002-2015, that was a lie! Your ears aren't protected, not at all - in fact, the lie that they are protected is so dangerous to ear protection that it gives *negative* ear protection!

Now, fixing them required adding functionality for volume multipliers to be assigned to head items. This isn't done /perfectly/, but I've done a lot of testing and it certainly works fine. If one would want to add volume_multipliers to other headwear after this PR, or var-edit it, they could do so, and I'd imagine the use cases for that are pretty big.

Also, I removed like five duplicate flags, too, but that's not really important to defeating the SPECTRE OF SERVICE-INDUCED HEARING LOSS.

SierraKomodo commented on code/modules/mob/living/carbon/human/human.dm:
```diff
@@ -1815,9 +1815,11 @@
+    . = ..()
+    for(var/obj/item/clothing/ears/C in list(l_ear, r_ear))
+        . = min(., C.volume_multiplier)
+    for(var/obj/item/clothing/head/D in list(head, l_ear, r_ear))
```
Why are you also checking l_ear and r_ear here? Head items only fit in the head slot.

gy1ta23 replied:
because i'm not a very good coder and i thought it didn't work without it. it does work without it, its irrelevant and i removed the second for loop thing so i could do the other suggested changes

SierraKomodo commented on code/modules/clothing/_clothing.dm:
```diff
@@ -392,6 +392,7 @@ BLIND     // can't see anything
 	var/light_overlay = "helmet_light"
 	var/light_applied
 	var/brightness_on
+	var/volume_multiplier = 1
```
Considering this is also present in ears, this var definition can probably just be moved to `/obj/item/clothing` and then overridden in /ear and /head.

gy1ta23 replied:
i am pretty sure i did what you want

gy1ta23 commented:
Should be all good now. Requested changes implemented

Spookerton commented on code/modules/mob/living/carbon/human/human.dm:
```diff
@@ -1815,9 +1815,9 @@
-	. = ..()
-	for(var/obj/item/clothing/ears/C in list(l_ear, r_ear))
-		. = min(., C.volume_multiplier)
+    . = ..()
+    for(var/obj/item/clothing/C in list(head, l_ear, r_ear))
+        . = min(., C.volume_multiplier)
```
```suggestion
	. = ..()
	for(var/obj/item/clothing/C in list(head, l_ear, r_ear))
		. = min(., C.volume_multiplier)
```
used spaces for indentation; must be tabs

gy1ta23 replied:
fixed

gy1ta23 committed:
adds volume handling shit for heads and removes flag spam




 removes extxraneous ear tatgs


removes lines, volume modifier for all


does the spook indent thing spook likes

Co-authored-by: Spookerton <spkrtn@pm.me>
